crates/spicepod/src/component/dataset.rs

```diff
@@ -78,7 +78,7 @@ impl Dataset {
         Dataset {
             from,
             name,
-            mode: None,
+            mode: Mode::default(),
             sql: None,
             sql_ref: None,
             params: Option::default(),
```

Comment:
not related to changes, that should be just `None`

Reply:
I made this not optional